example/poisson/poisson_driver.cpp

```diff
@@ -59,13 +59,47 @@ TaskCollection PoissonDriver::MakeTaskCollection(BlockList_t &blocks) {
   const int num_partitions = pmesh->DefaultNumPartitions();
   TaskRegion &solver_region = tc.AddRegion(num_partitions);
 
+  // make some storage for doing an Allreduce
+  reductions.resize(1);
+  // initialize it to zero
+  reductions[0].val = 0.0;
+
   for (int i = 0; i < num_partitions; i++) {
     // make/get a mesh_data container for the state
     auto &md = pmesh->mesh_data.GetOrAdd("base", i);
     auto &mdelta = pmesh->mesh_data.GetOrAdd("delta", i);
 
     TaskList &tl = solver_region[i];
 
+    // pass a pointer to the variable being reduced into
+    auto loc_red = tl.AddTask(none, poisson_package::SumMass<MeshData<Real>>, md.get(),
+                              &reductions[0].val);
+    // make it a regional dependency so dependent tasks can't execute until all lists do
+    // this
+    solver_region.AddRegionalDependencies(0, i, loc_red);
+    // start a non-blocking MPI_Iallreduce
+    TaskID start_global_reduce = tl.AddTask(
+        loc_red, &parthenon::AllReduce<Real>::StartReduce, &reductions[0], i, MPI_SUM);
+    // test the reduction until it completes
+    TaskID finish_global_reduce = tl.AddTask(
+        start_global_reduce, &parthenon::AllReduce<Real>::CheckReduce, &reductions[0], i);
+    solver_region.AddRegionalDependencies(1, i, finish_global_reduce);
```

Comment:
I guess not in this case, but in general the receive doesn't need to be in the same place as the send right? There could be intermediate work

Reply:
It does not.  In fact, even though they are written next to each other, the other non-dependent tasks in the list can (and do!) execute until the global reduction finishes.  in other words, even here there _is_ intermediate work.

Reply:
Excellent. :+1:. It might be worth emphasizing that in a comment.